packages/blockchain-wallet-v4-frontend/src/components/Navbar/MobileDropdown/Switch/index.tsx

```diff
@@ -35,21 +35,21 @@ const getSwitchBGColor = (state: SwitchLabelState) => (props: SwitchLabelProps)
 
   if (props.selected) {
     if (state === 'active') {
-      return colors.grey900
+      return PaletteColors['grey-900']
     }
     if (state === 'hover') {
-      return colors.grey800
+      return PaletteColors['grey-800']
     }
 
-    return colors.grey900
+    return PaletteColors['grey-900']
   }
 
   if (state === 'active') {
-    return colors.grey900
+    return PaletteColors['grey-900']
   }
 
   if (state === 'hover') {
-    return colors.grey800
+    return PaletteColors['grey-800']
   }
   return 'transparent'
```

Comment:
This is logic is super confussing - it seems like it always returns the same 3 colors, `PaletteColors['grey-900']` for `active`, `PaletteColors['grey-800']` for `hover` or `transparent`. Some logic can be merged here. 

1. If `props.disabled`, returns `transparent`
2. If `props.selected` and `state.active`, returns `PaletteColors['grey-900']`
3. If `props.selected` and `state.hover`, returns `PaletteColors['grey-800']`
4. If `props.selected` and state is not active nor hover, returns `PaletteColors['grey-900']`. Same as case 2.
5. If `!props.selected` and `state.active`, returns `PaletteColors['grey-900']`. Same as case 2.
6. If `!props.selected` and `state.hover`, returns `PaletteColors['grey-800']`. Same as case 3.
7. Else of all, returns `transparent`